src/widgets.tsx

```diff
@@ -1,132 +1,244 @@
 import * as React from "react";
 import * as ReactDOM from "react-dom";
+import { cell_utils, notebook_utils as nb_utils } from "./utils";
 import { Widget } from "@phosphor/widgets";
 import { VCSMenu, VCSMenuProps } from "./components/VCSMenu";
 import { CommandRegistry } from "@phosphor/commands";
 import { DocumentRegistry } from "@jupyterlab/docregistry";
-import { GET_VARS_CMD, LOAD_DEPENDENCIES } from "./constants";
+import { GET_VARS_CMD, CHECK_MODULES_CMD, READY_KEY } from "./constants";
+import { NotebookTracker, NotebookPanel } from "@jupyterlab/notebook";
+import { ThemeManager } from "@jupyterlab/apputils";
 
 export class LeftSideBarWidget extends Widget {
   div: HTMLDivElement; // The div container for this widget
   commands: CommandRegistry; // Jupyter app CommandRegistry
-  context: DocumentRegistry.Context; // Jupyter app DocumentRegistry.Context
-  notebook: any; // The notebook this widget is interacting with
+  notebook_panel: NotebookPanel; // The notebook this widget is interacting with
+  notebook_tracker: NotebookTracker; // This is to track current notebooks
   variables: string;
   component: any; // the LeftSidebar component
-  context_path: string;
   loading_data: boolean;
-
-  constructor(commands: CommandRegistry, context: DocumentRegistry.Context) {
+  current_file: string;
+  constructor(commands: CommandRegistry, tracker: NotebookTracker) {
     super();
     this.div = document.createElement("div");
     this.div.id = "left-sidebar";
     this.node.appendChild(this.div);
     this.commands = commands;
-    this.context = context;
-    this.notebook = null;
-    this.inject = this.inject.bind(this);
+    this.notebook_tracker = tracker;
+    this.notebook_panel = tracker.currentWidget;
     this.loading_data = false;
-  
-    if (this.context) {
-      this.context_path = this.context.session.name;
-    }
-
+    this.current_file = "";
+    this.inject = this.inject.bind(this);
+    this.getFilePath = this.getFilePath.bind(this);
```

Comment:
these should be declared as setters and getters. See https://stackoverflow.com/questions/12827266/get-and-set-in-typescript

Reply:
I think I did what you're asking, let me know